src/threads/infrastructure/persistence/relational/entities/thread.entity.ts

```diff
@@ -0,0 +1,23 @@
+import { Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+import { Thread } from '../../../../domain/thread';
+import { EntityRelationalHelper } from '../../../../../utils/relational-entity-helper';
+import { UserEntity } from '../../../../../users/infrastructure/persistence/relational/entities/user.entity';
+import { MessageEntity } from '../../../../../messages/infrastructure/presistence/entities/message.entity';
+
+@Entity({
+  name: 'thread',
+})
+export class ThreadEntity extends EntityRelationalHelper implements Thread {
+  @PrimaryGeneratedColumn()
+  id: number | string;
+
+  @ManyToOne(() => UserEntity, {
+    eager: true,
+  })
+  participants: UserEntity[];
+
+  @ManyToOne(() => MessageEntity, {
+    eager: true,
+  })
+  parentMessage: MessageEntity;
```

Comment:
The entity must have a composite primary key of participant and parent message, I think it can be like this:
```suggestion
export class ThreadParticipant {
  @PrimaryColumn()
  participantId: number;

  @PrimaryColumn()
  parentMessageId: number;

  @ManyToOne(() => ParticipantEntity)
  @JoinColumn({ name: 'participantId', referencedColumnName: 'id' })
  participant: UserEntity;

  @ManyToOne(() => MessageEntity)
  @JoinColumn({ name: 'parentMessageId', referencedColumnName: 'id' })
  parentMessage: MessageEntity;
}
```